starfish/pipeline/features/intensity_table.py

```diff
@@ -0,0 +1,260 @@
+    from_spot_data         creates an IntensityTable from a 3d array and a spot attributes dataframe
+    synthetic_intensities  creates an IntensityTable with synthetic spots, given a codebook
+
+    Methods
+    -------
+    save                   save the IntensityTable to netCDF
+    load                   load an IntensityTable from netCDF
+
+    Attributes
+    ----------
+    Constants.FEATURES     name of the first axis of the IntensityTable
+    Constants.GENE :       name of the field that stores the decoded gene identity for each feature
+    Constants.QUALITY      name of the field that stores the decoded gene quality for each feature
+    SpotAttributes.X       name of the pixelwise spot x-coordinate
+    SpotAttributes.Y       name of the pixelwise spot y-coordinate
+    SpotAttributes.Z       name of the pixelwise spot z-coordinate
+    SpotAttributes.RADIUS  name of the spot radius index
+
+    Examples
+    --------
+    >>> from starfish.util.synthesize import SyntheticData
+    >>> sd = SyntheticData(n_ch=3, n_hyb=4, n_codes=2)
+    >>> codes = sd.codebook()
+    >>> sd.intensities(codebook=codes)
+    <xarray.IntensityTable (features: 2, c: 3, h: 4)>
+    array([[[    0.,     0.,     0.,     0.],
+            [    0.,     0.,  8022., 12412.],
+            [11160.,  9546.,     0.,     0.]],
+
+           [[    0.,     0.,     0.,     0.],
+            [    0.,     0., 10506., 10830.],
+            [11172., 12331.,     0.,     0.]]])
+    Coordinates:
+      * features   (features) MultiIndex
+      - z          (features) int64 7 3
+      - y          (features) int64 14 32
+      - x          (features) int64 32 15
+      - r          (features) float64 nan nan
+      * c          (c) int64 0 1 2
+      * h          (h) int64 0 1 2 3
+        gene_name  (features) object 08b1a822-a1b4-4e06-81ea-8a4bd2b004a9 ...
+
+    """
+
+    class Constants(AugmentedEnum):
+        FEATURES = 'features'
+        GENE = 'gene_name'
+        QUALITY = 'quality'
+
+    class SpotAttributes(AugmentedEnum):
+        X = 'x'
+        Y = 'y'
+        Z = 'z'
+        RADIUS = 'r'
+
+    @classmethod
+    def empty_intensity_table(
+            cls, spot_attributes: pd.MultiIndex, n_ch: int, n_hyb: int
+    ) -> "IntensityTable":
+        """Create an empty intensity table with pre-set axis whose values are zero
+
+        Parameters
+        ----------
+        spot_attributes : pd.MultiIndex
+            MultiIndex containing spot metadata. Must contain the values specifid in Constants.X,
+            Y, Z, and RADIUS.
+        n_ch : int
+            number of channels measured in the imaging experiment
+        n_hyb : int
+            number of hybridization rounds measured in the imaging experiment
+
+        Returns
+        -------
+        IntensityTable :
+            empty IntensityTable
+
+        """
+        cls._verify_spot_attributes(spot_attributes)
+        channel_index = pd.Index(np.arange(n_ch))
```

Comment:
isn't the pd.Index in these two lines not necessary?

Reply:
I'll test and remove if that's the case.